Template Error when running common static analysis recipe

Got this error when running the common static analysis recipe on the repo https://github.com/spring-projects/spring-hateoas

<img width="1183" alt="Screen Shot 2023-02-27 at 10 53 18 AM" src="https://user-images.githubusercontent.com/122563761/221656423-f1cbe3aa-6f72-48dd-8064-4986061f566d.png">

Stack trace:
```stacktrace
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:4.40.0:dryRun (default-cli) on project spring-hateoas: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:4.40.0:dryRun failed: Exception while visiting project file 'src/main/java/org/springframework/hateoas/CollectionModel.java (in CollectionModel.deriveElementType)', caused by: java.lang.IllegalArgumentException: Expected a template that would generate exactly one statement to replace one statement, but generated 4. Template:
Objects::nonNull, at org.openrewrite.java.internal.template.JavaTemplateJavaExtension$1.maybeReplaceStatement(JavaTemplateJavaExtension.java:462)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:298)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:4.40.0:dryRun failed: Exception while visiting project file 'src/main/java/org/springframework/hateoas/CollectionModel.java (in CollectionModel.deriveElementType)', caused by: java.lang.IllegalArgumentException: Expected a template that would generate exactly one statement to replace one statement, but generated 4. Template:
Objects::nonNull, at org.openrewrite.java.internal.template.JavaTemplateJavaExtension$1.maybeReplaceStatement(JavaTemplateJavaExtension.java:462)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:298)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.openrewrite.internal.RecipeRunException: Exception while visiting project file 'src/main/java/org/springframework/hateoas/CollectionModel.java (in CollectionModel.deriveElementType)', caused by: java.lang.IllegalArgumentException: Expected a template that would generate exactly one statement to replace one statement, but generated 4. Template:
Objects::nonNull, at org.openrewrite.java.internal.template.JavaTemplateJavaExtension$1.maybeReplaceStatement(JavaTemplateJavaExtension.java:462)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:324)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:172)
    at org.openrewrite.java.JavaTemplate.withTemplate (JavaTemplate.java:107)
    at org.openrewrite.java.JavaTemplate.withTemplate (JavaTemplate.java:39)
    at org.openrewrite.java.tree.J.withTemplate (J.java:91)
    at org.openrewrite.java.cleanup.ReplaceLambdaWithMethodReference$1.visitLambda (ReplaceLambdaWithMethodReference.java:120)
    at org.openrewrite.java.cleanup.ReplaceLambdaWithMethodReference$1.visitLambda (ReplaceLambdaWithMethodReference.java:64)
    at org.openrewrite.java.tree.J$Lambda.acceptJava (J.java:2939)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.lambda$visitContainer$33 (JavaVisitor.java:1321)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitContainer (JavaVisitor.java:1321)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:856)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitReturn (JavaVisitor.java:1015)
    at org.openrewrite.java.tree.J$Return.acceptJava (J.java:4613)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4 (JavaVisitor.java:367)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitBlock (JavaVisitor.java:366)
    at org.openrewrite.java.tree.J$Block.acceptJava (J.java:761)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitMethodDeclaration (JavaVisitor.java:817)
    at org.openrewrite.java.tree.J$MethodDeclaration.acceptJava (J.java:3407)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4 (JavaVisitor.java:367)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitBlock (JavaVisitor.java:366)
    at org.openrewrite.java.tree.J$Block.acceptJava (J.java:761)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitClassDeclaration (JavaVisitor.java:454)
    at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava (J.java:1211)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10 (JavaVisitor.java:473)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitCompilationUnit (JavaVisitor.java:473)
    at org.openrewrite.java.JavaVisitor.visitJavaSourceFile (JavaVisitor.java:461)
    at org.openrewrite.java.tree.J$CompilationUnit.acceptJava (J.java:1514)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.RecipeScheduler.lambda$scheduleVisit$4 (RecipeScheduler.java:270)
    at org.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeScheduler.java:56)
    at org.openrewrite.scheduling.ForkJoinScheduler.lambda$schedule$0 (ForkJoinScheduler.java:41)
    at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec (ForkJoinTask.java:1375)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:165)
Caused by: java.lang.IllegalArgumentException: Expected a template that would generate exactly one statement to replace one statement, but generated 4. Template:
Objects::nonNull
    at org.openrewrite.java.internal.template.JavaTemplateJavaExtension$1.maybeReplaceStatement (JavaTemplateJavaExtension.java:462)
    at org.openrewrite.java.internal.template.JavaTemplateJavaExtension$1.visitLambda (JavaTemplateJavaExtension.java:266)
    at org.openrewrite.java.internal.template.JavaTemplateJavaExtension$1.visitLambda (JavaTemplateJavaExtension.java:59)
    at org.openrewrite.java.tree.J$Lambda.acceptJava (J.java:2939)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:172)
    at org.openrewrite.java.JavaTemplate.withTemplate (JavaTemplate.java:107)
    at org.openrewrite.java.JavaTemplate.withTemplate (JavaTemplate.java:39)
    at org.openrewrite.java.tree.J.withTemplate (J.java:91)
    at org.openrewrite.java.cleanup.ReplaceLambdaWithMethodReference$1.visitLambda (ReplaceLambdaWithMethodReference.java:120)
    at org.openrewrite.java.cleanup.ReplaceLambdaWithMethodReference$1.visitLambda (ReplaceLambdaWithMethodReference.java:64)
    at org.openrewrite.java.tree.J$Lambda.acceptJava (J.java:2939)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.lambda$visitContainer$33 (JavaVisitor.java:1321)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitContainer (JavaVisitor.java:1321)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:856)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.visitMethodInvocation (JavaVisitor.java:849)
    at org.openrewrite.java.tree.J$MethodInvocation.acceptJava (J.java:3665)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitReturn (JavaVisitor.java:1015)
    at org.openrewrite.java.tree.J$Return.acceptJava (J.java:4613)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4 (JavaVisitor.java:367)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitBlock (JavaVisitor.java:366)
    at org.openrewrite.java.tree.J$Block.acceptJava (J.java:761)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitMethodDeclaration (JavaVisitor.java:817)
    at org.openrewrite.java.tree.J$MethodDeclaration.acceptJava (J.java:3407)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1268)
    at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4 (JavaVisitor.java:367)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitBlock (JavaVisitor.java:366)
    at org.openrewrite.java.tree.J$Block.acceptJava (J.java:761)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.visitClassDeclaration (JavaVisitor.java:454)
    at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava (J.java:1211)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:356)
    at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10 (JavaVisitor.java:473)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:145)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:126)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:145)
    at org.openrewrite.java.JavaVisitor.visitCompilationUnit (JavaVisitor.java:473)
    at org.openrewrite.java.JavaVisitor.visitJavaSourceFile (JavaVisitor.java:461)
    at org.openrewrite.java.tree.J$CompilationUnit.acceptJava (J.java:1514)
    at org.openrewrite.java.tree.J.accept (J.java:64)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:276)
    at org.openrewrite.RecipeScheduler.lambda$scheduleVisit$4 (RecipeScheduler.java:270)
    at org.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeScheduler.java:56)
    at org.openrewrite.scheduling.ForkJoinScheduler.lambda$schedule$0 (ForkJoinScheduler.java:41)
    at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec (ForkJoinTask.java:1375)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:165)
```

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/ReplaceLambdaWithMethodReferenceTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/ReplaceLambdaWithMethodReferenceTest.java
@@ -16,7 +16,6 @@
 
 package org.openrewrite.java.cleanup;
 
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.openrewrite.Issue;
 import org.openrewrite.java.JavaIsoVisitor;
@@ -507,7 +506,6 @@ class ReplaceLambdaWithMethodReferenceTest implements RewriteTest {
         );
     }
 
-    @Disabled
     @Issue("https://github.com/openrewrite/rewrite/issues/2897")
     @Test
     void notNullToObjectsNonNullError() {
@@ -542,6 +540,7 @@ class ReplaceLambdaWithMethodReferenceTest implements RewriteTest {
               """,
             """
               import java.util.Collection;
+              import java.util.Objects;
 
               public class A {
                   private static Class<?> determineCommonAncestor(Class<?> clazz1, Class<?> clazz2) {

--- a/rewrite-java/src/main/java/org/openrewrite/java/internal/template/BlockStatementTemplateGenerator.java
+++ b/rewrite-java/src/main/java/org/openrewrite/java/internal/template/BlockStatementTemplateGenerator.java
@@ -116,8 +116,21 @@ public class BlockStatementTemplateGenerator {
                     @SuppressWarnings("unchecked") J2 t = (J2) tree;
 
                     if (blockEnclosingTemplateComment != null) {
+                        boolean caughtStopComment = false;
+                        if (getCursor().getParent() != null && getCursor().getParent().getValue() instanceof JContainer) {
+                            JContainer container = getCursor().getParent().getValue();
+                            if (container.getBefore() != null && container.getBefore().getComments() != null) {
+                                for (Comment comment : container.getBefore().getComments()) {
+                                    if (comment instanceof TextComment && ((TextComment) comment).getText().equals(STOP_COMMENT)) {
+                                        caughtStopComment  =true;
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+
                         //noinspection unchecked
-                        J2 trimmed = (J2) TemplatedTreeTrimmer.trimTree(t);
+                        J2 trimmed = caughtStopComment ? null : (J2) TemplatedTreeTrimmer.trimTree(t);
                         if (trimmed != null) {
                             js.add(trimmed);
                         } else {

